app/controllers/ViewAllMovementsController.scala

```diff
@@ -16,74 +16,110 @@
       renderView(Ok, ern, searchOptions)
     }
+  }
 
   def onSubmit(ern: String, searchOptions: MovementListSearchOptions): Action[AnyContent] =
     authorisedWithData(ern).async { implicit request =>
       formProvider().bindFromRequest().fold(
-        _ => renderView(BadRequest, ern, searchOptions),
+        formWithErrors => renderView(BadRequest, ern, searchOptions, formWithErrors),
         value => Future(Redirect(routes.ViewAllMovementsController.onPageLoad(ern, value)))
       )
     }
 
-  private def renderView(status: Status, ern: String, searchOptions: MovementListSearchOptions)(implicit request: DataRequest[_]) = {
-    connector.getMovementList(ern, Some(searchOptions)).map {
-      case Right(movementList) =>
+  private def renderView(
+                          status: Status,
+                          ern: String,
+                          searchOptions: MovementListSearchOptions,
+                          form: Form[MovementListSearchOptions] = formProvider()
+                        )(implicit request: DataRequest[_]): Future[Result] = {
+
+    val result: EitherT[Future, ErrorResponse, Result] = for {
+      movementList <- EitherT(getMovementListConnector.getMovementList(ern, Some(searchOptions)).map {
+        case Left(NotFoundError) => Right(GetMovementListResponse(Seq(), 0))
+        case value => value
+      })
+
+      epcs <- EitherT(getExciseProductCodesConnector.getExciseProductCodes())
+      exciseProductCodeOptions = MovementListSearchOptions.CHOOSE_PRODUCT_CODE +: epcs
+
+      countries <- EitherT(getMemberStatesConnector.getMemberStates())
+      sortedCountries = MovementListSearchOptions.CHOOSE_COUNTRY +: countries.sortBy(_.displayName)
+    } yield {
+
+      val pageCount: Int = calculatePageCount(movementList)
 
-        val pageCount = {
-          if (movementList.count == 0) {
-            1
-          } else if (movementList.count % DEFAULT_MAX_ROWS != 0) {
-            (movementList.count / DEFAULT_MAX_ROWS) + 1
-          } else {
-            movementList.count / DEFAULT_MAX_ROWS
-          }
-        }
-        if (searchOptions.index <= 0 || searchOptions.index > pageCount) {
-          Redirect(routes.ViewAllMovementsController.onPageLoad(ern, MovementListSearchOptions()))
-        } else {
-          status(view(
-            form = formProvider().fill(searchOptions),
-            action = routes.ViewAllMovementsController.onSubmit(ern, searchOptions),
-            ern = ern,
-            movements = movementList.movements,
-            sortSelectItems = MovementSortingSelectOption.constructSelectItems(Some(searchOptions.sortBy.code)),
-            searchSelectItems = MovementSearchSelectOption.constructSelectItems(searchOptions.searchKey.map(_.code)),
-            pagination = paginationHelper.constructPagination(pageCount, ern, searchOptions)
-          ))
-        }
+      if (searchOptions.index <= 0 || searchOptions.index > pageCount) {
```

Comment:
Is it worth a small comment explaining why this guard is here, and the reason for the redirect?